### PR TITLE
on_message_reaction listeners

### PR DESCRIPTION
**Pull request by Wolfhound905**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition 

## Description

Adds listeners for reaction events. Example usage.
```py
@listen()
async def on_message_reaction_remove_all(event):
    print(event.message.content)

@listen()
async def on_message_reaction_remove(event):
    print(event.emoji.name)

@listen()
async def on_message_reaction_add(event):
    print(event.emoji.name)
```
add and remove returns the CustomEmoji class.
remove_all returns message class

## Changes

- Adds reaction events

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.9.x`
- [x] I've tested my code

### CODE REVIEW

File: dis_snek/client.py [hunk 1]
```diff
@@ -825,6 +826,52 @@ async def _on_raw_message_update(self, event: RawGatewayEvent) -> None:
...
+            event: raw reaction add event
+        """
+        if member := event.data.get("member"):
+            author = self.cache.place_member_data(event.data.get("guild_id"), member)
+        else:
+            author = await self.cache.get_user(event.data.get("user_id"))
+
+        emoji = CustomEmoji.from_dict(event.data.get("emoji"), self)
+
+        self.dispatch(events.MessageReactionAdd(message=None, emoji=emoji, author=author))
```
Comment (benwoo1110) [inline dis_snek/client.py:844]: Why is `message=None` here? I think either we change to only pass the `message_id` or get the full message from the cache.

File: dis_snek/client.py [hunk 2]
```diff
@@ -825,6 +826,52 @@ async def _on_raw_message_update(self, event: RawGatewayEvent) -> None:
...
+            event: raw reaction remove event
+        """
+        if member := event.data.get("member"):
+            author = self.cache.place_member_data(event.data.get("guild_id"), member)
+        else:
+            author = await self.cache.get_user(event.data.get("user_id"))
+
+        emoji = CustomEmoji.from_dict(event.data.get("emoji"), self)
+
+        self.dispatch(events.MessageReactionRemove(message=None, emoji=emoji, author=author))
```
Comment (benwoo1110) [inline dis_snek/client.py:861]: Same comment as above

File: dis_snek/client.py [hunk 3]
```diff
@@ -825,6 +826,52 @@ async def _on_raw_message_update(self, event: RawGatewayEvent) -> None:
...
+
+        Args:
+            event: raw reaction remove event
+        """
+        if member := event.data.get("member"):
+            author = self.cache.place_member_data(event.data.get("guild_id"), member)
+        else:
+            author = await self.cache.get_user(event.data.get("user_id"))
+
+        emoji = CustomEmoji.from_dict(event.data.get("emoji"), self)
```
Comment (benwoo1110) [inline dis_snek/client.py:859]: The gateway only returns partial emoji data, I think should use the `Emoji` class instead.

**Review by benwoo1110 — CHANGES_REQUESTED** [review]
Left some comments. Also, have you tested them to be working?

File: dis_snek/client.py (hunk 1 above)
Comment (Wolfhound905) [reply dis_snek/client.py:844]: I have not tried sending just the message_id. I can at another time.

**Pull request closed without merging**